File: result/metagpt/claude/humaneval/7.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Function to filter an array of strings based on whether they contain a given substring
char** filter_by_substring(char** strings, int num_strings, const char* substring, int* out_num_strings) {
    // Allocate memory for the output array (maximum size is num_strings)
    char** out = (char**)malloc(num_strings * sizeof(char*));
    int count = 0;

    // Iterate through each string in the input array
    for (int i = 0; i < num_strings; i++) {
        // Check if the substring is present in the current string
        if (strstr(strings[i], substring) != NULL) {
            // If found, add the string to the output array
            out[count] = strings[i];
            count++;
        }
    }

    // Set the number of strings in the output array
    *out_num_strings = count;

    // Return the filtered array
    return out;
}

File: result/metagpt/gpt4o/humaneval/7.rs
fn filter_by_substring(strings: &[&str], substring: &str) -> Vec<&str> {
    let mut out = Vec::new();

    for &string in strings {
        if string.contains(substring) {
            out.push(string);
        }
    }

    out
}

fn main() {
    let strings = vec!["hello", "world", "hell", "hello world"];
    let substring = "hell";
    let filtered_strings = filter_by_substring(&strings, substring);
    println!("Filtered strings: {:?}", filtered_strings);
}